pkg/tests/tasks/migration/migration_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2024 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migration
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/app"
+	"github.com/maistra/maistra-test-tool/pkg/tests/ossm"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/curl"
+	"github.com/maistra/maistra-test-tool/pkg/util/ns"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
+)
+
+func TestMigrationSimpleClusterWide(t *testing.T) {
+	test.NewTest(t).MinVersion(version.SMCP_2_6).Groups(test.Full).Run(func(t test.TestHelper) {
+		t.Cleanup(func() {
+			oc.DeleteResource(t, "", "Istio", "example")
+			oc.DeleteResource(t, "", "IstioCNI", "default")
+		})
+		defaults := ossm.DefaultClusterWideSMCP()
+		ossm.BasicSetup(t)
+		templ := `apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: {{ .Name }}
+spec:
+  version: {{ .Version }}
+  tracing:
+    type: None
+  security:
+    manageNetworkPolicy: false
+  gateways:
+    enabled: false
+  policy:
+    type: Istiod
+  addons:
+    grafana:
+      enabled: false
+    kiali:
+      enabled: false
+    prometheus:
+      enabled: false
+  mode: ClusterWide`
+		oc.ApplyTemplate(t, defaults.Namespace, templ, defaults)
+
+		oc.WaitSMCPReady(t, defaults.Namespace, defaults.Name)
+		oc.WaitSMMRReady(t, defaults.Namespace)
+		oc.Label(t, "", "Namespace", ns.Bookinfo, "istio-injection=enabled")
+		t.LogStep("Update SMMR to include bookinfo namespace")
+		// Wait for SMMR to include bookinfo
+		oc.DefaultOC.WaitFor(t, defaults.Namespace, "ServiceMeshMemberRoll", "default", `jsonpath='{.status.configuredMembers[?(@=="bookinfo")]}'`)
+
+		t.LogStep("Create 3.0 controlplane and IstioCNI")
+		istio := `apiVersion: sailoperator.io/v1alpha1
+kind: Istio
+metadata:
+  name: example
+spec:
+  namespace: istio-system
+  version: v1.24.1`
```

Comment:
This will change often so maybe it should be parametrized and passed via env variable?

Reply:
Yes, something similar as `SMCP_VERSION` which is there

Reply:
also, I am thinking, do we even need to explicitly specify the istio/istio CNI version here ( in matt) ? When we don't specify them, it will be automatically picked by the operator (whichever we install in the pipeline), won't it?